.github/workflows/build.yml

```diff
@@ -1,41 +1,50 @@
 name: Build
 
-on: [push, pull_request]
+on:
+  push:
+    tags:
+      - "*"
+  workflow_dispatch:
 
 jobs:
-    build:
-        runs-on: ${{ matrix.os }}
-        strategy:
-            matrix:
-                os: [windows-latest, ubuntu-latest]
-                node-version: [14]
-        steps:
-            - name: Check out Git repository
-              uses: actions/checkout@v3
-
-            - name: Install Node.js, NPM and Yarn
-              uses: actions/setup-node@v3
-              with:
-                  node-version: ${{ matrix.node-version }}
-                  cache: "npm"
-
-            - name: Install dependencies
-              run: npm ci
-
-            - name: Type check
-              run: npm run type-check
-
-            - name: Lint check
-              run: npm run lint-check
-
-            - name: Format check
-              run: npm run format-check
-
-            - name: Electron Builder Action
-              uses: samuelmeuli/action-electron-builder@v1.6.0
-              env:
-                  NODE_OPTIONS: "--max_old_space_size=8192"
-              with:
-                  github_token: ${{ secrets.github_token }}
-                  args: "--config .electron-builder.config.js"
-                  release: ${{ startsWith(github.ref, 'refs/tags/v') }}
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest]
+        node-version: [14]
```

Comment:
iirc, the reason I had to revert back to node 14 when attempting to upgrade this project to node 18 some time ago was because the previous electron-builder action plugin broke with it. as you've removed the reliance on that plugin, I'm gonna try node 18 again here and see how it goes

Reply:
actually I'll do that separately, gonna merge this